bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -421,6 +421,7 @@ private ForceWriteRequest(Handle<ForceWriteRequest> recyclerHandle) {
 
         private void recycle() {
             logFile = null;
+            flushed = false;
```

Comment:
This kind of bug is completely avoided by removing object recycling. I understand that object recycling was added a while ago, but I wonder if it is still necessary on JVM 17. It'd be valuable to run benchmarks to verify the underlying assumption that we get better performance by recycling objects.

Reply:
This is a great point

Reply:
It should be easy to test because you can disable the recycler using Netty Java properties